pennylane/math/single_dispatch.py

```diff
@@ -315,6 +318,18 @@ def _transpose_tf(a, axes=None):
 
 ar.register_function("tensorflow", "transpose", _transpose_tf)
 
+
+def _eigvalsh(density_matrix):
+    evs = _i("tf").linalg.eigvalsh(density_matrix)
+    evs = _i("tf").math.real(evs)
+    return evs
+
+
+ar.register_function("tensorflow", "eigvalsh", _eigvalsh)

```

Comment:
Just to prevent any possible headaches in the future 😄 
```suggestion
def _eigvalsh_tf(density_matrix):
    evs = _i("tf").linalg.eigvalsh(density_matrix)
    evs = _i("tf").math.real(evs)
    return evs


ar.register_function("tensorflow", "eigvalsh", _eigvalsh_tf)
```